library/src/main/kotlin/de/adesso/junitinsights/junit/JUnitCallbacks.kt

```diff
@@ -39,7 +47,16 @@ class JUnitCallbacks :
     /**
      * Gets called after the complete test plan has been executed, so the report can be generated.
      */
-    override fun testPlanExecutionFinished(testPlan: TestPlan) = EventLog.writeReport()
+    override fun testPlanExecutionFinished(testPlan: TestPlan) {
+        val report = reportCreator.createReport(getReportPageTitle(), EventLog.events)
+        reportWriter.writeReport(report)
+    }
+
+    private fun getReportPageTitle(): String {
```

Comment:
This method should be a part of the `ReportCreator`. The `Report` has the attribute `created`, which should contain the same date that is written in the title, so the filename can match the title later on.

The `created` attribute of `Report` should also be used in the `ReportWriter` for that matter.

Reply:
The attribute is used in `ReportWriter`. I have taken the opportunity and refactored some of the parts in that class

Reply:
Okay, to clarify what I meant with this:

- I expect the date that appears in the filename to be the exact same date as the one in the page title
- In `ReportCreator::createReport`, you create a new `Report` object and pass the **then current date** to the `created` property of the constructor, because you call the `Date()` function
- The `created` property is used later on in `ReportWriter::getReportFileName` to generate the filename
- In `ReportCreator::getReportPageTitle` you generate the page title for the report using the **then current date** because you call `Date()` **again**
- This means that the two dates are **not identical** which could lead to confusion

Suggestion: Remove the `projectName` property from the `Report` class and generate the page title in the `ReportWriter`

Reply:
I changed the code so that the date is only generated once. That value is now passed to the creation of the project name (the title of the report page) and also as a prop to the report.

I kept the `projectName` prop in `Report` because this is not an issue with the data structure.

Another thing: Should we change the property name from "project name" to "report title" or something like that? As it includes the date and time, "project name" is not really correct.
That would require changes in both the backend and the frontend.

Reply:
What do we need the `projectName` (or `reportTitle`) property for? In my opinion it is totally unnecessary and we should remove it alltogether because the report title is created the same way every time which can also be done in the ReportWriter.